metrics/README.md

```diff
@@ -25,3 +25,100 @@ function in that file has the ability to also `curl` or `socat` the JSON results
+1. Clone `cloud-native-setup` repository into a preferred directory and change directory upto `cloud-native-setup/metrics`:
+   ```sh
+   $ git clone https://github.com/clearlinux/cloud-native-setup.git
+   $ cd cloud-native-setup/metrics
+   ```
+2. Launch the execution by:
+   ```sh
+   $ ./scaling/k8s_scale.sh
+   INFO: Initialising
+   command: bc: yes
+   command: jq: yes
+   INFO: Checking k8s accessible
+   INFO: 1 k8s nodes in 'Ready' state found
+   starting kubectl proxy
+   Starting to serve on 127.0.0.1:8090
+   daemonset.apps/stats created
+   Waiting for daemon set "stats" rollout to finish: 0 of 1 updated pods are available...
+   daemon set "stats" successfully rolled out
+   INFO: Running test
+   INFO: And grab some stats
+   INFO: idle [98.49] free [29031100] launch [0] node [clr-30f01b5149ba4ab8b05a7ee03b6812a5] inodes_free [31103039]
+   INFO: Testing replicas 1 of 20
+   INFO: Content of runtime_command=:/@RUNTIMECLASS@/d
+   ...
+   ```
+The above execution might take about 4min because it launch upto 20 pods by default and takes measurements for CPU utilization, memory utilization and pod boot time, finally it will generate a `k8s-scaling.json` result file at `result` directory.
+
+**Note**: by default scaling framework makes call to the kubernetes API directly so, if facing conectivity issues verify that `kubelet` service's proxies and `no_proxy` environment variable are properly setup.
+
+**Note**: by default scaling framework uses default values for all its required variables, which can be checked through `scaling/k8s_scale.sh -h` and updated when launching the execution, i.e.:
+```
+$ ./scaling/k8s_scale.sh -h
+Usage: ./scaling/k8s_scale.sh [-h] [options]
+   Description:
+	Launch a series of workloads and take memory metric measurements after
+	each launch.
+   Options:
+		-h,    Help page.
+
+Environment variables:
+	Name (default)
+		Description
+	TEST_NAME (k8s scaling)
+		Can be set to over-ride the default JSON results filename
+	NUM_PODS (20)
+		Number of pods to launch
+	STEP (1)
+		Number of pods to launch per cycle
+	wait_time (30)
+		Seconds to wait for pods to become ready
+	delete_wait_time (600)
+		Seconds to wait for all pods to be deleted
+	settle_time (5)
+		Seconds to wait after pods ready before taking measurements
+	use_api (yes)
+		specify yes or no to use the API to launch pods
+	grace (30)
+		specify the grace period in seconds for workload pod termination
+
+$ use_api=no ./scaling/k8s_scale.sh
+```
+
+The steps to generate the result report are listed below:
+
+1. Having the `result/k8s-scaling.json` result file, create a directory into `results` with a preferred name and place `k8s-scaling.json` file into it, so the file distribution looks like:
+   ```sh
+   $ tree result
+   results/
+   └── scaling
+       └── k8s-scaling.json
+   ```
+2. Launch the report generation by:
+   ```sh
+   ./report/makereport.sh
```

Comment:
Note, the first time you run this script it will build a docker container to generate the reports and this process can take several minutes. Subsequent runs will be much faster.

Reply:
Done.